maven-plugin/src/main/java/org/openmrs/maven/plugins/Pull.java

```diff
@@ -0,0 +1,293 @@
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.transport.URIish;
+import org.openmrs.maven.plugins.model.Server;
+import org.openmrs.maven.plugins.utility.Project;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Set;
+
+/**
+ * @goal pull
+ * @requiresProject false
+ */
+public class Pull extends AbstractTask {
+
+    private static final String NO_WATCHED_MODULES_MESSAGE = "Server with id %s has no watched modules";
+    private static final String PULL_SUCCESS_MESSAGE = "Module %s have been updated successfully";
+    private static final String PULL_ERROR_MESSAGE = "Module %s could not be updated automatically due to %s";
+    private static final String CONFLICT_MESSAGE = "conflicts";
+    private static final String UPDATE_MANUALLY_MESSAGE = "Changes need to be pulled manually\n";
+    private static final String UPDATING_MODULE_MESSAGE = "Module %s is being updated to the latest upstream master";
+    private static final String CREATING_LOCAL_REPO_ERROR = "problem when initializing local repository";
+    private static final String GIT_COMMAND_ERROR = "problem with resolving git command";
+    private static final String CREATING_TEMP_REMOTE_ERROR = "problem with creating temporary remote to perform automatic update";
+
+    /**
+     * @parameter expression="${serverId}"
+     */
+    private String serverId;
+
+    private String failureReason;
+
+    public Pull() {}
+
+    public Pull(AbstractTask other) {super(other);}
+
+    public Pull(MavenProject project, MavenSession session, BuildPluginManager manager) {
+        super.mavenProject = project;
+        super.mavenSession = session;
+        super.pluginManager = manager;
+    }
+
+    @Override
+    public void executeTask() throws MojoExecutionException, MojoFailureException {
+        serverId = wizard.promptForExistingServerIdIfMissing(serverId);
+        Server server = Server.loadServer(serverId);
+
+        if(server.hasWatchedProjects()){
+            boolean pullingError = false;
+            Set<Project> watchedProjects = server.getWatchedProjects();
+            for(Project project: watchedProjects) {
+                wizard.showMessage(String.format(UPDATING_MODULE_MESSAGE, project.getArtifactId()));
+                if (pullLatestUpstream(project.getPath())) {
+                    wizard.showMessage(String.format(PULL_SUCCESS_MESSAGE, project.getArtifactId()));
+                } else {
+                    getLog().error(String.format(PULL_ERROR_MESSAGE, project.getArtifactId(), failureReason));
+                    getLog().error(UPDATE_MANUALLY_MESSAGE);
+                    pullingError = true;
+                }
+            }
+            if(pullingError){
+                throw new MojoExecutionException("Some modules could not be updated automatically, please see error messages above");
```

Comment:
I'd rather we list all modules that were updated and all that were not updated here.